apps/trading/client-pages/swap/swap-container.tsx

```diff
@@ -0,0 +1,307 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import {
+  useAssetsMapProvider,
+  type AssetFieldsFragment,
+} from '@vegaprotocol/assets';
+import {
+  getBaseAsset,
+  getQuoteAsset,
+  isSpot,
+  marketDataProvider,
+  useMarketsMapProvider,
+  type MarketDataFieldsFragment,
+  type MarketFieldsFragment,
+} from '@vegaprotocol/markets';
+import { VegaIcon, VegaIconNames, Link } from '@vegaprotocol/ui-toolkit';
+import { useVegaWallet } from '@vegaprotocol/wallet-react';
+import { useAccounts, type Account } from '@vegaprotocol/accounts';
+import { useT } from '../../lib/use-t';
+import {
+  addDecimal,
+  addDecimalsFormatNumber,
+  formatNumber,
+  removeDecimal,
+} from '@vegaprotocol/utils';
+import { OrderTimeInForce, OrderType, Side } from '@vegaprotocol/types';
+import { useVegaTransactionStore } from '@vegaprotocol/web3';
+import noop from 'lodash/noop';
+import { getNotionalSize } from '@vegaprotocol/deal-ticket';
+import { useDataProvider } from '@vegaprotocol/data-provider';
+import { AssetInput, SwapButton, PriceImpactInput } from './swap-form';
+import BigNumber from 'bignumber.js';
+
+const POLLING_TIME = 5000;
+
+const assetBalance = (
+  asset?: AssetFieldsFragment,
+  accounts?: Account[] | null
+) => {
+  if (!asset || !accounts) return undefined;
+  const account = accounts.find((a) => a.asset.id === asset.id);
+  return account?.balance;
+};
+
+export interface SwapFields {
+  baseId: string;
+  quoteId: string;
+  baseAmount: string;
+  quoteAmount: string;
+  priceImpactTolerance: string;
+}
+
+const derivePrice = (
+  marketData: MarketDataFieldsFragment | null,
+  side: Side,
+  toleranceFactor: number,
+  market: MarketFieldsFragment
```

Comment:
The `market` argument doesn't appear to be used within `derivePrice`

Reply:
i will add back the tick size logic that uses it

Reply:
added back, for the tick size logic